clang/lib/AST/Type.cpp

```diff
@@ -5150,6 +5152,9 @@ bool Type::isHLSLIntangibleType() const {
   CXXRecordDecl *RD = RT->getAsCXXRecordDecl();
   assert(RD != nullptr &&
          "all HLSL structs and classes should be CXXRecordDecl");
+
+  if (!RD->isCompleteDefinition())
+    return false;
   assert(RD->isCompleteDefinition() && "expecting complete type");
```

Comment:
Why leave the assert? A comment explaining how you know an incomplete definition is not an intangible type would be useful.

Reply:
Reverted this. Moved the check to the deduceAddressSpace function.
To be honest, I'm not completely sure why we don't get the definition when instantiating a RWBuffer. I'd believe we should get the definition.
I added a FIXME, and AST tests to cover it so we can move forward with the rest while all this is evolving.